src/Internal/Payment/Router.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Class Router
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WCPay\Core\Server\Request\Get_Payment_Process_Factors;
+use WCPay\Database_Cache;
+use WCPay\Exceptions\API_Exception;
+use WCPay\Internal\Payment\Factor;
+
+/**
+ * Until the new payment process is fully developed, and the legacy
+ * process is gone, the new process is managed as a feature behind a router.
+ *
+ * This class will be removed once the new process is the default
+ * option for all payments.
+ */
+class Router {
+	/**
+	 * Database cache.
+	 *
+	 * @var Database_Cache
+	 */
+	protected $database_cache;
+
+	/**
+	 * Class constructor, receiving dependencies.
+	 *
+	 * @param Database_Cache $database_cache Database cache.
+	 */
+	public function __construct( Database_Cache $database_cache ) {
+		$this->database_cache = $database_cache;
+	}
+
+	/**
+	 * Checks whether a given payment should use the new payment process.
+	 *
+	 * @param array $factors Factors, describing the type and conditions of the payment.
+	 * @return bool
+	 */
+	public function should_use_new_payment_process( array $factors ): bool {
+		$allowed = $this->get_enabled_factors();
```

Comment:
I'll switch to `allowed`, I think it's better for this context.

> I think enabled should be replaced with present because that's what it is - it's a factor that's part of the ongoing payment process.

Actually, this particular line is exactly for the allowed factors. `$factors` (the parameter) is about the present factors.

Reply:
Switched to `allowed` in this commit: https://github.com/Automattic/woocommerce-payments/pull/7035/commits/d3d5f2077d3d020c2e2ffc4b8fa5e7d109b95a27